packages/text-annotator/src/SelectionHandler.ts

```diff
@@ -122,6 +132,7 @@ export const SelectionHandler = (
     lastPointerDown = { ...evt, target, timeStamp, offsetX, offsetY, type };
 
     isLeftClick = evt.button === 0;
+    currentTarget = undefined;
```

Comment:
Having the `currentTarget` cleaned up on the `pointerdown` event guarantees that on the deletion check, we won't accidentally refer to the dismissed annotation

Reply:
UPD: It led to a regression while using the touch devices. When user scrolls while having an active selection range, they cannot proceed selecting anymore

Reply:
Another madness piece... In Firefox, the `selectstart` event gets dispatched AFTER the `selectionstart` 🤯 
![image](https://github.com/user-attachments/assets/23bb6215-5db1-4d8e-9482-feb955f9d3a0)